core-data/reducer.js

```diff
@@ -82,8 +82,29 @@ export function postTypes( state = {}, action ) {
 	return state;
 }
 
+/**
+ * Reducer managing indexes data.
+ *
+ * @param {Object} state  Current state.
+ * @param {Object} action Dispatched action.
+ *
+ * @return {Object} Updated state.
+ */
+export function indexData( state = {}, action ) {
```

Comment:
Continuing discussion from https://github.com/WordPress/gutenberg/pull/6318#discussion_r183690619 : That "index" is considered ambiguous doesn't seem to be helped by the addition of a suffix, and makes me wonder if we can better communicate what it is we're tracking. I guess the endpoint itself is properly called the index ([source](https://github.com/WordPress/WordPress/blob/41a82d60787ae739d15b8fd5309cf7387658733c/wp-includes/rest-api/class-wp-rest-server.php#L1008-L1052)). From some of the comments, we could derive other names like `siteIndex`, `siteData`, or maybe just `site`.

Most worrying to me is that `routes` is a very large data shape that we presumably don't really need here.